libs/wingsdk/src/target-tf-aws/function.ts

```diff
@@ -146,7 +149,8 @@ export class Function extends cloud.FunctionBase {
         : Duration.fromMinutes(1).seconds,
     });
 
-    this.arn = this.function.arn;
+    this.arn = this.function.qualifiedArn;
```

Comment:
Let’s change the arm that gets used in _bind to use the unqualified arn to avoid undesired recreations for each version

Reply:
Yea you are right, I thought we used the instance variable arn in the _bind method. I think this is actually really confusing since we are basically inverting the exposed arns between terraform resource Function and our AWS function.

Maybe just my mental model but I think I would want to deal with the unqualified arn more often/by-default over the qualified arn. Perhaps instead do

```typescript
this.arn = this.function.arn
this.qualifiedArn = this.function.arn
```

@eladb @Chriscbr any strong feelings/thoughts here?

Reply:
It is confusing to users or or maintainers? externally we only expose "arn" and "unqualifiedArn"..

See the discussion above, we wanted to help users to the right thing most times, which is to use the qualified ARN. If we have `arn` they will just use it.

Here's an alternative suggestion: we can expose `qualifiedArn` and `unqualifiedArn` and then it will require the user to understand these concepts.

Not sure that's a better solution because it adds cognitive friction where it's not really needed

Reply:
I think it’s confusing to both users and maintainers. By definition the qualified arn is the arn + suffix, and unqualified is just arn with no suffix. So I think to expose the qualified arn as just arn could lead to some misunderstandings. 

I’m also unsure (thinking out loud) that using the qualified arn is desired in most instances, I imagine most users will only care about invoking latest version of lambda code which the unqualified arn will always point to latest.

Reply:
Okay makes sense. Let's just leave it with ARN and qualifiedArn

Reply:
@raywonkari - I believe what they mean is if you could add a @returns docstring (annotation in the comments) over arn and qualifiedArn to explain what they are.